Objects/listobject.c

```diff
@@ -2580,7 +2583,10 @@ list_count(PyListObject *self, PyObject *value)
     Py_ssize_t i;
 
     for (i = 0; i < Py_SIZE(self); i++) {
-        int cmp = PyObject_RichCompareBool(self->ob_item[i], value, Py_EQ);
+        PyObject *obj = self->ob_item[i];
+        Py_INCREF(obj);
+        int cmp = PyObject_RichCompareBool(obj, value, Py_EQ);
```

Comment:
How about add the refcount in PyObject_RichCompareBool()?

Reply:
Managing the reference count is the responsibility of the caller, not of `PyObject_RichCompareBool`.